metricsbp/baseplate_hooks.go

```diff
@@ -101,3 +106,76 @@ var (
 	_ tracing.CreateChildSpanHook  = (*spanHook)(nil)
 	_ tracing.AddSpanCounterHook   = (*spanHook)(nil)
 )
+
+// NewConcurrencyCreateSpanHook returns a CreateServerSpanHook that can be used
+// to monitor the number of concurrent requests being processed and spawns a
+// reporter gorouting to publish that as a gauge.
+//
+// The reporter will update the gauge at the given interval.
+// The reporter goroutine will be stopped when the metrics context is canceled.
+// The gauge will be named "runtime.{hostname}.PID{process_id}.active_requests".
+func NewConcurrencyCreateSpanHook(metrics *Statsd, interval time.Duration) tracing.CreateServerSpanHook {
+	var activeRequests int64
+	counter := &atomicCounter{count: &activeRequests}
+	hostname, err := os.Hostname()
+	if err != nil {
+		hostname = "UNKOWN-HOSTNAME"
+	}
+	gauge := metrics.Gauge(
+		fmt.Sprintf("runtime.%s.PID%d.active_requests", hostname, os.Getpid()),
+	)
+	go func() {
+		for {
+			time.Sleep(interval)
```

Comment:
don't use `time.Sleep` in a for loop, use `time.Tick`/`time.Ticker` instead.

but I think this should be handled by [`RunSysStats`](https://pkg.go.dev/github.com/reddit/baseplate.go/metricsbp?tab=doc#Statsd.RunSysStats) function, using same prefixes and stuff (we are actually using labels for that one, and I think we should still use labels for this one as well).